src/site/content/en/blog/constructable-stylesheets/index.md

```diff
@@ -0,0 +1,181 @@
+## Constructing a stylesheet
+
+Rather than introducing a new API to accomplish this, the [Constructable
+StyleSheets](https://drafts.csswg.org/cssom/#dom-cssstylesheet-cssstylesheet)
+specification makes it possible to create stylesheets imperatively by invoking
+the `CSSStyleSheet()` constructor. The resulting CSSStyleSheet object has two
+new methods that make it safer to add and update stylesheet rules without
+triggering [Flash of Unstyled
+Content](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) (FOUC).
+The
+[`replace()`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/replace)
+and
+[`replaceSync()`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/replaceSync)
+methods both replace the stylesheet with a string of CSS, and `replace()`
+returns a Promise. In both cases, external stylesheet references are not
+supported—any `@import` rules are ignored and will produce a warning.
+
+```js
+const sheet = new CSSStyleSheet();
+
+// replace all styles synchronously:
+sheet.replaceSync('a { color: red; }');
+
+// replace all styles:
+sheet.replace('a { color: blue; }')
+  .then(() => {
+    console.log('Styles replaced');
+  })
+  .catch(err => {
+    console.error('Failed to replace styles:', err);
+  });
+
+// Any @import rules are ignored.
+// Both of these still apply the a{} style:
+sheet.replaceSync('@import url("styles.css"); a { color: red; }');
+sheet.replace('@import url("styles.css"); a { color: red; }');
+// Console warning: "@import rules are not allowed here..."
+```
+
+{% Aside 'gotchas' %}
+In earlier versions of the specification,
+`replace()` allowed `@import` rules and returned
+a Promise that resolved when these were finished loading. This feature was
+[removed from the specification](https://github.com/WICG/construct-stylesheets/issues/119#issuecomment-642300024)
+and `@import` rules are
+ignored with a warning as of Chrome 84.
+{% endAside %}
+
+## Using constructed stylesheets
+
+The second new feature introduced by Constructable StyleSheets is an
+[adoptedStyleSheets](https://drafts.csswg.org/cssom/#extensions-to-the-document-or-shadow-root-interface)
+property available on [Shadow
+Roots](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM)
+and [Documents](https://developer.mozilla.org/en/docs/Web/API/Document). This
+lets us explicitly apply the styles defined by a `CSSStyleSheet` to a given DOM
+subtree. To do so, we set the property to an array of one or more stylesheets to
+apply to that element.
+
+```js
+// Create our shared stylesheet:
+const sheet = new CSSStyleSheet();
+sheet.replaceSync('a { color: red; }');
+
+// Apply the stylesheet to a document:
+document.adoptedStyleSheets = [sheet];
+
+// Apply the stylesheet to a Shadow Root:
+const node = document.createElement('div');
+const shadow = node.attachShadow({ mode: 'open' });
+shadow.adoptedStyleSheets = [sheet];
+```
+
+Notice that we're overriding the value of `adoptedStyleSheets` instead of
```

Comment:
Note that the landed spec actually says that `adoptedStyleSheets` is defined in terms of `ObservableArray`, which means that it is no longer frozen. You should be able to do `shadow.adoptedStyleSheets.push(sheet);`. Having said that, no implementation (yet) does it this way. AFAIK, Gecko might be the first to ship this new behavior. I'm hoping Chromium is close to doing that also. But in the meantime, perhaps the right thing to do is to leave this content as-is, so that it works in browsers? I just wanted to call it out.